internal/rate/storage.go

```diff
@@ -0,0 +1,58 @@
+package rate
+
+import (
+	"crypto/tls"
+	"time"
+
+	"github.com/redis/go-redis/v9"
+
+	"github.com/TykTechnologies/tyk/config"
+)
+
+// NewStorage provides a redis v9 client for rate limiter use.
+func NewStorage(cfg *config.StorageOptionsConf) redis.UniversalClient {
+	// poolSize applies per cluster node and not for the whole cluster.
+	poolSize := 500
+	if cfg.MaxActive > 0 {
+		poolSize = cfg.MaxActive
+	}
+
+	timeout := 5 * time.Second
+
+	if cfg.Timeout > 0 {
+		timeout = time.Duration(cfg.Timeout) * time.Second
+	}
+
+	var tlsConfig *tls.Config
+
+	if cfg.UseSSL {
+		tlsConfig = &tls.Config{
+			InsecureSkipVerify: cfg.SSLInsecureSkipVerify,
+		}
+	}
+
+	opts := &redis.UniversalOptions{
+		Addrs:            cfg.HostAddrs(),
+		MasterName:       cfg.MasterName,
+		SentinelPassword: cfg.SentinelPassword,
+		Username:         cfg.Username,
+		Password:         cfg.Password,
+		DB:               cfg.Database,
+		DialTimeout:      timeout,
+		ReadTimeout:      timeout,
+		WriteTimeout:     timeout,
+		//		IdleTimeout:      240 * timeout,
+		PoolSize:  poolSize,
+		TLSConfig: tlsConfig,
+	}
+
```

Comment:
@titpetric any chance to interface it? so we can expose a redis.UniversalOptions interface and you can import it here. @mativm02 wdyt about it?

The main idea is to avoid the importing of `go-redis/v9` directly and encourage the use of the storage library

Reply:
@sredxny @mativm02 I've left a `Client() redis.UniversalClient` in storage which would be good enough to replace this constructor.

It's likely we could have a `(*config.StorageOptionsConf).RedisUniversalOptions() (*redis.UniversalOptions, error)` which would give us a few things:

1. source of truth for anything config logic = config package,
2. validation of configs done in config package,
3. ability to interface the function providing the config for mocks/fakes

We could also use a more generic interface to redis to decouple from redis itself over time (e.g. the internal storage drivers idea where a single interface would be v8 and v9.... not returning *redis.StringCmd and similar package-bound returns)

Reply:
Also if we don't do it this way we can be delaying the release for this since some of the algos require v9 client

Reply:
Resolving this to merge and iterate when new storage is ready.